### PR TITLE
TIG-2915 Allow parameters to be passed into genny from the caller

### PR DESCRIPTION
**Pull request by aggrand**
Also evaluate parameters before storing them, to avoid recursive parameter evaluation.

Still manually checking some of the workload yamls but they seem equivalent so far.

### CODE REVIEW

File: src/lamplib/src/tests/test_preprocess.py [hunk 1]
```diff
@@ -359,7 +359,76 @@ def test_bad_external_state_4(self):
...
+    def test_load_entire_workload_nested_param(self):
+        yaml_input = """SchemaVersion: 2018-07-01
+
+LoadConfig:
+    Path: "src/testlib/configs/workload.yml"
+    Parameters:
+        Repeat: 3
+"""
+
+        # Even though the nested workload defines a "Repeat"
```
Comment (vrachev) [inline src/lamplib/src/tests/test_preprocess.py:420]: 👍

**Review by vrachev — APPROVED** [review]
LGTM! Nice clean change

**Comment by aggrand** [comment]
evergreen merge

**Comment by aggrand** [comment]
evergreen merge

**Comment by aggrand** [comment]
Failing due to the known benchmark bug, squash-merging.

**Pull request merged**